### ISSUE TITLE
Shape error when using object detection API

### ISSUE THREAD

**Issue by hanzy123** [tensorflow/models#2096]
My training always interrupted after few steps,The output is here:
```
INFO:tensorflow:Scale of 0 disables regularizer.
INFO:tensorflow:Scale of 0 disables regularizer.
INFO:tensorflow:Scale of 0 disables regularizer.
INFO:tensorflow:Scale of 0 disables regularizer.
INFO:tensorflow:Scale of 0 disables regularizer.
INFO:tensorflow:Scale of 0 disables regularizer.
INFO:tensorflow:Summary name Learning Rate is illegal; using Learning_Rate instead.
INFO:tensorflow:Summary name /clone_loss is illegal; using clone_loss instead.
/root/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/gradients_impl.py:93: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "
2017-08-02 10:31:13.608186: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
2017-08-02 10:31:13.608209: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
2017-08-02 10:31:13.608214: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
2017-08-02 10:31:13.608218: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
2017-08-02 10:31:13.608222: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
2017-08-02 10:31:13.712671: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:901] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2017-08-02 10:31:13.713150: I tensorflow/core/common_runtime/gpu/gpu_device.cc:887] Found device 0 with properties:
name: GeForce GTX 1080
major: 6 minor: 1 memoryClockRate (GHz) 1.8475
pciBusID 0000:01:00.0
Total memory: 7.92GiB
Free memory: 7.81GiB
2017-08-02 10:31:13.713179: I tensorflow/core/common_runtime/gpu/gpu_device.cc:908] DMA: 0
2017-08-02 10:31:13.713186: I tensorflow/core/common_runtime/gpu/gpu_device.cc:918] 0:   Y
2017-08-02 10:31:13.713193: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX 1080, pci bus id: 0000:01:00.0)
INFO:tensorflow:Restoring parameters from object_detection/Human_detection/output/model.ckpt-0
2017-08-02 10:31:22.372886: I tensorflow/core/common_runtime/simple_placer.cc:669] Ignoring device specification /device:GPU:0 for node 'prefetch_queue_Dequeue' because the input edge from 'prefetch_queue' is a reference connection and already has a device field set to /device:CPU:0
INFO:tensorflow:Starting Session.
INFO:tensorflow:Starting Queues.
INFO:tensorflow:global_step/sec: 0
INFO:tensorflow:global step 1: loss = 1.6937 (37.98 sec/step)
2017-08-02 10:33:23.330706: I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 2749 get requests, put_count=2117 evicted_count=1000 eviction_rate=0.472367 and unsatisfied allocation rate=0.630047
2017-08-02 10:33:23.330754: I tensorflow/core/common_runtime/gpu/pool_allocator.cc:259] Raising pool_size_limit_ from 100 to 110
INFO:tensorflow:global step 2: loss = 1.9736 (0.32 sec/step)
INFO:tensorflow:global step 3: loss = 1.6014 (0.87 sec/step)
INFO:tensorflow:global step 4: loss = 1.3160 (0.35 sec/step)
INFO:tensorflow:global step 5: loss = 1.0027 (0.42 sec/step)
INFO:tensorflow:global step 6: loss = 0.6841 (0.28 sec/step)
INFO:tensorflow:global step 7: loss = 0.5660 (0.83 sec/step)
INFO:tensorflow:global step 8: loss = 0.5574 (0.71 sec/step)
INFO:tensorflow:global step 9: loss = 0.4658 (0.25 sec/step)
INFO:tensorflow:global step 10: loss = 0.4297 (0.27 sec/step)
INFO:tensorflow:global step 11: loss = 0.3972 (0.29 sec/step)
INFO:tensorflow:global step 12: loss = 0.3471 (0.93 sec/step)
2017-08-02 10:33:28.848779: W tensorflow/core/framework/op_kernel.cc:1152] Invalid argument: Incompatible shapes: [1,62,4] vs. [1,64,4]
	 [[Node: gradients/Loss/BoxClassifierLoss/Loss/sub_grad/BroadcastGradientArgs = BroadcastGradientArgs[T=DT_INT32, _device="/job:localhost/replica:0/task:0/gpu:0"](gradients/Loss/BoxClassifierLoss/Loss/sub_grad/Shape, gradients/Loss/BoxClassifierLoss/Loss/sub_grad/Shape_1)]]
2017-08-02 10:33:28.850346: W tensorflow/core/framework/op_kernel.cc:1152] Invalid argument: Incompatible shapes: [1,62,4] vs. [1,64,4]
	 [[Node: gradients/Loss/BoxClassifierLoss/Loss/sub_grad/BroadcastGradientArgs = BroadcastGradientArgs[T=DT_INT32, _device="/job:localhost/replica:0/task:0/gpu:0"](gradients/Loss/BoxClassifierLoss/Loss/sub_grad/Shape, gradients/Loss/BoxClassifierLoss/Loss/sub_grad/Shape_1)]]
2017-08-02 10:33:28.850387: W tensorflow/core/framework/op_kernel.cc:1152] Invalid argument: Incompatible shapes: [1,62,4] vs. [1,64,4]
	 [[Node: gradients/Loss/BoxClassifierLoss/Loss/sub_grad/BroadcastGradientArgs = BroadcastGradientArgs[T=DT_INT32, _device="/job:localhost/replica:0/task:0/gpu:0"](gradients/Loss/BoxClassifierLoss/Loss/sub_grad/Shape, gradients/Loss/BoxClassifierLoss/Loss/sub_grad/Shape_1)]]
INFO:tensorflow:Error reported to Coordinator: <class 'tensorflow.python.framework.errors_impl.InvalidArgumentError'>, Incompatible shapes: [1,62,4] vs. [1,64,4]
	 [[Node: gradients/Loss/BoxClassifierLoss/Loss/sub_grad/BroadcastGradientArgs = BroadcastGradientArgs[T=DT_INT32, _device="/job:localhost/replica:0/task:0/gpu:0"](gradients/Loss/BoxClassifierLoss/Loss/sub_grad/Shape, gradients/Loss/BoxClassifierLoss/Loss/sub_grad/Shape_1)]]
```

**Comment by shivaniag** [comment]
Please provide details about what platform you are using  (operating system, architecture). Also include your TensorFlow version. Also, did you compile from source or install a binary?  Make sure you also include the exact command if possible to produce  the output included in your test case. If you are unclear what to include  see the issue template displayed in  [the Github new issue template](https://github.com/tensorflow/tensorflow/issues/new).

 We ask for this in the issue submission template, because    it is really difficult to help without that information. Thanks!

**Comment by hanzy123** [comment]
My platform is Ubuntu16.04 and using Nvidia-smi_375.66,My Cuda version is Cuda8.0.My tensorflow version is 1.1.0

**Comment by hanzy123** [comment]
I have the same error as #1618 but i can't deal it after clearing all checkpoint as he said

**Comment by hanzy123** [comment]
I fixed the problem.On my side,I made bad `image/object/class/text` and `image/object/class/label`.After I correct the `text` and `label`, the problem gone

**Comment by Sharathnasa** [comment]
@hanzy123 Facing the same issue. What exactly did you changed?

**Comment by hanzy123** [comment]
Maybe you have the bad training data,In my case, i set the classes_text = 1, classes =1 at the begining and get the error.After i set the classes_text=[1,1,1,...,1] and classes = [1,1,1,...,1] ,It worked!

**Comment by auroua** [comment]
I encountered the same error. I found the error caused by wrongly defining the tfrecords data structures.
I wrongly write the key like this:
```
 'image/object/box/class/text': dataset_util.bytes_list_feature(classes_text),
 'image/object/box/class/label': dataset_util.int64_list_feature(classes),
``` 
When I changed to the following, the error disappears.
```
 'image/object/class/text': dataset_util.bytes_list_feature(classes_text),
 'image/object/class/label': dataset_util.int64_list_feature(classes),
```

**Comment by Manish-rai21bit** [comment]
I have this issue. This is the error I'm getting. I believe that this is because of some bad training data shapes. I was wondering how to narrow down on which parameter is giving this error.

`InvalidArgumentError (see above for traceback): Assign requires shapes of both tensors to match. lhs shape= [9] rhs shape= [273]
	 [[Node: save/Assign_2 = Assign[T=DT_FLOAT, _class=["loc:@BoxPredictor_0/ClassPredictor/biases"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](BoxPredictor_0/ClassPredictor/biases, save/RestoreV2:2)]]`

**Comment by polololya** [comment]
I faced this issue when I resized images (made them 50% smaller) but didn't modify xml annotations accordingly
